cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java

```diff
@@ -97,13 +106,59 @@ private Builder(Schema schema) {
      * @throws UnexpectedFormatException if the field is not in the schema, or the field is not nullable but a null
      *                                   value is given.
      */
-    public Builder set(String fieldName, Object value) {
+    public Builder set(String fieldName, @Nullable Object value) {
       validateAndGetField(fieldName, value);
       fields.put(fieldName, value);
       return this;
     }
 
     /**
+     * Convert the given date into the type of the given field, and set the value for that field.
+     * A Date can be converted into a long or a string.
+     *
+     * @param fieldName Name of the field to set.
+     * @param date Date value for the field.
+     * @return This builder.
+     * @throws UnexpectedFormatException if the field is not in the schema, or the field is not nullable but a null
+     *                                   value is given, or the date cannot be converted to the type for the field.
+     */
+    public Builder convertAndSet(String fieldName, @Nullable Date date) throws UnexpectedFormatException {
+      return convertAndSet(fieldName, date, DEFAULT_FORMAT);
+    }
+
+    /**
+     * Convert the given date into the type of the given field, and set the value for that field.
+     * A Date can be converted into a long or a string.
+     *
+     * @param fieldName Name of the field to set.
```

Comment:
A Date can be converted into a long or a string, using a format string, if supplied.